apps/prairielearn/src/models/enrollment.ts

```diff
@@ -1,16 +1,60 @@
+import { Response } from 'express';
 import { loadSqlEquiv, queryOptionalRow, queryRow } from '@prairielearn/postgres';
-import { Enrollment, EnrollmentSchema } from '../lib/db-types';
+import error = require('@prairielearn/error');
+
+import { CourseInstance, Enrollment, EnrollmentSchema, Institution } from '../lib/db-types';
+import { isEnterprise } from '../lib/license';
+import { checkEnterpriseEnrollment } from '../ee/models/enrollment';
 
 const sql = loadSqlEquiv(__filename);
 
-export async function insertEnrollment({
+export async function ensureEnrollment({
   course_instance_id,
   user_id,
 }: {
   course_instance_id: string;
   user_id: string;
 }): Promise<Enrollment> {
-  return await queryRow(sql.insert_enrollment, { course_instance_id, user_id }, EnrollmentSchema);
+  return await queryRow(sql.ensure_enrollment, { course_instance_id, user_id }, EnrollmentSchema);
+}
+
+export async function insertCheckedEnrollment(
```

Comment:
Add a jsDoc function description to explain what the return value means?

Reply:
Done in 4ccf5ba.